Adding loki logs

gaantunes opened the pull request:
This PR adds 6 panels for logs using Loki datasource. I am taking two different log sources here:

1. 3 panels (syslog errors, kernel logs and auth log) being targeted to logs scraped directly from the log files, using promtail.
2. 3 other panels (syslog, kernel, stdout) being targeted to logs scraped from journal. 

I decided to add those 2 sources for some reasons:

Journal gives us many labels out-of-the-box, including an unit label for stdout which categorizes the log producer systemd application, giving the user the possibility to navigate logs from a specific application. It is not every application that uses systemd, though. That`s why the first group of panels includes one targeted at syslog errors, which should be useful if this applications is writing to stdout, directing it to syslog by default.

I am also replicating the kernel logs, in case the user don`t want to give the Grafana Agent access permissions directly to the file.

The auth information is very important and I could not find a way to get it from journal, so it is going to be exclusive for direct scraping, as long as me or someone finds out a way to do it.

Right now I am only changing the Node dahsboard, as soon as those changes are approved I will also include it into the USE Method dashboard.

I also inluded a job template variable, so that if there are many job templates the user can select which one he wants.

Here are some screenshots:

![image](https://user-images.githubusercontent.com/9431850/154546323-b6b50ca8-372c-4bfd-b276-bfc63c53897f.png)
![image](https://user-images.githubusercontent.com/9431850/154546400-94ca858d-1ad0-4b9f-b57e-97dd9c9f0fe7.png)

rgeyer commented on docs/node-mixin/dashboards/node.libsonnet:
```diff
@@ -224,6 +281,30 @@ local gauge = promgrafonnet.gauge;
+          },
+          hide: 0,
+          label: 'Loki Data Source',
+          name: 'logs_datasource',
```
We're stumbling toward a consensus that a dashboard with multiple datasources should name them unambiguously, with their type. I.E. This should be..

```suggestion
          name: 'loki_datasource',
```

Further, the metrics datasource should be renamed `prometheus_datasource`.

rgeyer reviewed (changes requested):
This looks great, and the overall direction makes sense. These log panels will add a lot to the linux node dashboards.

As we discussed in another forum, it would be best to make the addition of these new template vars, and panels an opt-it configuration flag. That should make it easier to accept and merge upstream, and provides some flexibility.

gaantunes commented:
Additional edits:

- replaced a reference to [github.com/kubernetes-monitoring/kubernetes-mixin/lib/promgrafonnet/promgrafonnet.libsonnet](http://github.com/kubernetes-monitoring/kubernetes-mixin/lib/promgrafonnet/promgrafonnet.libsonnet) that was being used for gaugePanel, in favor of the [github.com/grafana/grafonnet-lib/grafonnet-7.0/grafana.libsonnet](http://github.com/grafana/grafonnet-lib/grafonnet-7.0/grafana.libsonnet) lib
- Added the conditional for Loki logs to both dashboards
- Following our new convention for datasource names

rgeyer commented on docs/node-mixin/dashboards/node.libsonnet:
```diff
@@ -4,253 +4,393 @@ local row = grafana.row;
+
+        $._NodeDashboard
+        .addTemplate(lokiDatasourceTemplate)
+        .addTemplate(jobTemplate)      
```
Is this ever used? I don't see any place that the job template var is referenced in a query.

While the linter *does* have this as a rule, there's no sense in having the template var if it doesn't impact the output. Feel free to remove it, and add a `.lint` file to ignore the job rule. I may actually need to improve the linter in this area.

gaantunes replied:
Yeah this is something I was thinking about, maybe the instance variable by itself is enough for what we need, which is correlating logs and metrics.

gaantunes replied:
It is used though the `nodeExporterSelector` variable on the queries. I was thinking about removing it, since the instance variable is enough for our goals. But it is inhenrited from the upstream mixin, so I kept it and plan to configure it as `job="$job"` on the integration, this way if the user wants to have multiple jobs he can.

gaantunes replied:
Right now it is fixed at `job="integrations/node_exporter"` on our integration.

rgeyer reviewed (approved):
LGTM, one nit.
